starter-code/public/javascript/index.js

```diff
@@ -1,23 +1,90 @@
-const charactersAPI = new APIHandler('http://localhost:8000');
+// const charactersAPI = new APIHandler('http://localhost:8000');
 
 window.addEventListener('load', () => {
   document.getElementById('fetch-all').addEventListener('click', function (event) {
+    axios.get('http://localhost:8000/characters').then(result => {
+      let resultsFromAPI = result.data
+      console.log(resultsFromAPI)
+      resultsFromAPI.forEach(character => {
+        let newNode = document.getElementsByClassName('character-info')[0].cloneNode(true)
 
-  });
+        newNode.getElementsByClassName('name')[0].innerText = character.name
+        newNode.getElementsByClassName('occupation')[0].innerText = character.occupation
+        newNode.getElementsByClassName('cartoon')[0].innerText = character.cartoon
+        newNode.getElementsByClassName('weapon')[0].innerText = character.weapon
 
-  document.getElementById('fetch-one').addEventListener('click', function (event) {
+        document.getElementsByClassName('characters-container')[0].append(newNode)
+      })
 
+    })
```

Comment:
its also good to do some error handling
this will help you in project module to find your error faster and easie
, so here you should write

.catch(error => console.log("something with grtting charecter id went wrong", error))

Reply:
@numbing Ah I see what you mean. Just added the error messages. Thanks a lot for the feedback. That's very helpful :)